src/hierarchies/nnig_hierarchy.h

```diff
@@ -15,14 +15,18 @@
 #include "base_hierarchy.h"
 #include "likelihoods/uni_norm_likelihood.h"
 #include "priors/nig_prior_model.h"
-#include "updaters/nnig_updater.h"
+// #include "updaters/nnig_updater.h"
+#include "updaters/mala_updater.h"
+#include "updaters/random_walk_updater.h"
 
 class NNIGHierarchy : public BaseHierarchy<NNIGHierarchy, UniNormLikelihood,
-                                           NIGPriorModel, NNIGUpdater> {
+                                           NIGPriorModel, MalaUpdater> {
  public:
-  NNIGHierarchy() = default;
   ~NNIGHierarchy() = default;
 
+  using BaseHierarchy<NNIGHierarchy, UniNormLikelihood, NIGPriorModel,
+                      MalaUpdater>::BaseHierarchy;
+
   bayesmix::HierarchyId get_id() const override {
     return bayesmix::HierarchyId::NNIG;
   }
```

Comment:
Sure, actually it's something I'd like to discuss. Maybe we don't need to template over the updater, since they actually have the same API. We can just have a pointer to the abstract one. In this way, it's easy to switch between different updaters to compare performances, and we can have a "default" one in a constructor.

Reply:
> Sure, actually it's something I'd like to discuss. Maybe we don't need to template over the updater, since they actually have the same API. We can just have a pointer to the abstract one. In this way, it's easy to switch between different updaters to compare performances, and we can have a "default" one in a constructor.

According to the last changes, this is by far the best solution. Since we managed to preserve the same API in every updater (from every specific Gibbs, to MALA and MH) we can remove it as template parameter.